pkg/sdk/testint/sequences_integration_test.go

```diff
@@ -0,0 +1,167 @@
+package testint
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+ * todo: `ALTER SEQUENCE [ IF EXISTS ] <name> UNSET COMMENT` not works, and error: Syntax error: unexpected 'COMMENT'. (line 39)
+ */
+
+func TestInt_Sequences(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	databaseTest, schemaTest := testDb(t), testSchema(t)
+
+	cleanupSequenceHandle := func(t *testing.T, id sdk.SchemaObjectIdentifier) func() {
+		t.Helper()
+		return func() {
+			err := client.Sequences.Drop(ctx, sdk.NewDropSequenceRequest(id))
+			if errors.Is(err, sdk.ErrObjectNotExistOrAuthorized) {
+				return
+			}
+			require.NoError(t, err)
+		}
+	}
+
+	createSequenceHandle := func(t *testing.T) *sdk.Sequence {
+		t.Helper()
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, random.StringN(4))
+		sr := sdk.NewCreateSequenceRequest(id).WithStart(sdk.Int(1)).WithIncrement(sdk.Int(1))
+		err := client.Sequences.Create(ctx, sr)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSequenceHandle(t, id))
+
+		s, err := client.Sequences.ShowByID(ctx, id)
+		require.NoError(t, err)
+		return s
+	}
+
+	t.Run("create sequence", func(t *testing.T) {
+		name := random.StringN(4)
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, name)
+
+		comment := random.StringN(4)
+		request := sdk.NewCreateSequenceRequest(id).
+			WithWith(sdk.Bool(true)).
+			WithStart(sdk.Int(1)).
+			WithIncrement(sdk.Int(1)).
+			WithIfNotExists(sdk.Bool(true)).
+			WithValuesBehavior(sdk.ValuesBehaviorPointer(sdk.ValuesBehaviorOrder)).
+			WithComment(&comment)
+		err := client.Sequences.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSequenceHandle(t, id))
```

Comment:
can we run showbyid after creation to check?

Reply:
done